File: simplified-app-shared/src/main/java/org/nypl/simplified/app/reader/ReaderTOCSelectionListenerType.java
package org.nypl.simplified.app.reader;

import org.nypl.simplified.app.reader.ReaderTOC.TOCElement;
import org.nypl.simplified.books.core.BookmarkAnnotation;

/**
 * A listener that receives the results of the TOC selection:
 * A TOC Item or a Bookmark Annotation.
 */

public interface ReaderTOCSelectionListenerType
{
  /**
   * A TOC item was selected.
   *
   * @param e The selected item
   */

  void onTOCSelectionReceived(TOCElement e);

  /**
   * A Bookmark annotation was selected.
   *
   * @param bm The selected bookmark
   */

  void onBookmarkSelectionReceived(BookmarkAnnotation bm);
}

File: simplified-viewer-epub-readium1/src/main/java/org/nypl/simplified/viewer/epub/readium1/toc/ReaderTOCSelectionListenerType.kt
package org.nypl.simplified.viewer.epub.readium1.toc

interface ReaderTOCSelectionListenerType {

  /**
   * Something was selected from the TOC.
   *
   * @param selection The selected item
   */

  fun onTOCItemSelected(selection: ReaderTOCSelection)
}
